core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -54,7 +54,8 @@ abstract class AbstractFetcherThread(name: String,
                                      val sourceBroker: BrokerEndPoint,
                                      failedPartitions: FailedPartitions,
                                      fetchBackOffMs: Int = 0,
-                                     isInterruptible: Boolean = true)
+                                     isInterruptible: Boolean = true,
+                                     val replicaManager: ReplicaManager)
```

Comment:
Really unfortunate to introduce this dependence. Could we just pass through `BrokerTopicStats` instead?

Reply:
ok, makes sense